org.eclipse.scanning.example/src/org/eclipse/scanning/example/detector/RandomLineDevice.java

```diff
@@ -103,10 +129,64 @@ public boolean write(IPosition pos) throws ScanningException {
 		return true;
 	}
 
+	protected void count(StackTraceElement[] ste) throws ScanningException {
+		count(ste, null);
+	}
+	protected void count(StackTraceElement[] ste, Object value) throws ScanningException {
+		String methodName = getMethodName(ste);
+		Integer count = counts.get(methodName);
+		if (count==null) count = 0;
+		count = count+1;
+		counts.put(methodName, count);
+		if (!values.containsKey(methodName)) values.put(methodName, new ArrayList<>());
+		try {
+			values.get(methodName).add(value!=null?BeanUtils.cloneBean(value):null);
+		} catch (Exception e) {
+			throw new ScanningException("Cannot clone information during test", e);
+		}
+	}
+	
+	public int getCount(String method) {
+		if (!counts.containsKey(method)) return 0;
+		return counts.get(method);
+	}
+	
+	public Object getValue(String method, int index) {
+		return values.get(method).get(index);
+	}
+	
+	protected static final String getMethodName ( StackTraceElement ste[] ) {  
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the array designator from the variable to the type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1197)